.circleci/config.yml

```diff
@@ -109,6 +101,74 @@ services_steps: &services_steps
     - store_test_results:
         path: junit
 
+run_package_tests: &run_package_tests
+  when: always
+  command: |
+    # https://discuss.circleci.com/t/switch-nodejs-version-on-machine-executor-solved/26675/3
+    set +e
+    export NVM_DIR="/opt/circleci/.nvm"
+    [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+    nvm install $NODE_VERSION
+    nvm use $NODE_VERSION
+    node --version
+    npm run test:package
+
+package_steps: &package_steps
+  steps:
+    - checkout
+
+    - restore_cache:
+        keys:
+          - v2-dependencies-{{ checksum "package-lock.json" }}
+          # https://github.com/badges/shields/issues/1937
+          - v2-dependencies-
+
+    - run:
+        name: Install dependencies
+        command: |
+          set +e
+          export NVM_DIR="/opt/circleci/.nvm"
+          [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+          nvm install v12
+          nvm use v12
+          npm install -g npm
+          npm ci
+
+    - run:
+        <<: *run_package_tests
+        environment:
+          mocha_reporter: mocha-junit-reporter
+          MOCHA_FILE: junit/gh-badges/results.xml
+          NODE_VERSION: v8
+        name: Run package tests on Node 8
+
+    - run:
+        <<: *run_package_tests
+        environment:
+          mocha_reporter: mocha-junit-reporter
+          MOCHA_FILE: junit/gh-badges/results.xml
+          NODE_VERSION: v10
+        name: Run package tests on Node 10
+
+    - run:
+        <<: *run_package_tests
+        environment:
+          mocha_reporter: mocha-junit-reporter
+          MOCHA_FILE: junit/gh-badges/results.xml
+          NODE_VERSION: v11
+        name: Run package tests on Node 11
+
+    - run:
+        <<: *run_package_tests
+        environment:
+          mocha_reporter: mocha-junit-reporter
+          MOCHA_FILE: junit/gh-badges/results.xml
```

Comment:
Could you use a different directory for each test run? e.g. `junit/gh-badges/v12/results.xml`